src/storage.rs

```diff
@@ -1169,20 +1174,16 @@ pub fn determine_best_parents(db: &Connection, joint: &Joint, witnesses: &[Strin
       level-witnessed_level ASC, \
       unit ASC \
     LIMIT 1",
-        parent_units,witness_list,COUNT_WITNESSES - MAX_WITNESS_LIST_MUTATIONS
+        parent_units,witness_list,witness_list_unit,COUNT_WITNESSES - MAX_WITNESS_LIST_MUTATIONS
     );
 
     let mut stmt = db.prepare(&sql)?;
     let rows = stmt
-        .query_map(&[&joint.unit.witness_list_unit], |row| row.get(0))?
+        .query_map(&[], |row| row.get(0))?
```

Comment:
here should use `Result<String>`, we can sure that if the unit returns `None` then the whole sql returns no lines.
and in the end just return `Ok(rows.into_iter().take(1))` to avoid unnecessary clone

Reply:
when i use `rows.into_iter().take(1)`, the compiler tells me found type `std::iter::Take<std::vec::IntoIter<std::string::String>>`,so i change to `nth(0)`